cas_object/src/cas_object_format.rs

```diff
@@ -1132,6 +1132,43 @@ impl CasObject {
         Ok((byte_offset_start, byte_offset_end))
     }
 
+    /// given a valid chunk_index, returns the uncompressed chunk length for the chunk
+    /// at the given index, chunk_index must be less than the number of chunks in the xorb
+    pub fn chunk_length(&self, chunk_index: u32) -> Result<u32, CasObjectError> {
+        self.validate_cas_object_info()?;
+        let chunk_index = chunk_index as usize;
+        if chunk_index >= self.info.unpacked_chunk_offsets.len() {
+            return Err(CasObjectError::InvalidArguments);
+        }
+        let cumulative_sum = self.info.unpacked_chunk_offsets[chunk_index];
+        let before = match chunk_index {
+            0 => 0,
+            _ => self.info.unpacked_chunk_offsets[chunk_index - 1],
+        };
+        Ok(cumulative_sum - before)
+    }
+
+    /// given a valid start and end, returns the uncompressed end-exclusive range length
+    /// meaning the length of the chunk range [chunk_index_start, chunk_index_end)
+    /// the following condition must be valid: chunk_index_end < chunk_index_end <= num_chunks
+    pub fn uncompressed_range_length(
+        &self,
+        chunk_index_start: u32,
+        chunk_index_end: u32,
+    ) -> Result<u32, CasObjectError> {
+        self.validate_cas_object_info()?;
+        if chunk_index_start >= chunk_index_end || chunk_index_end > self.info.num_chunks {
```

Comment:
Minor but arguably chunk_index_start == chunk_index_end just means a length of 0 rather than error.